<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Miguel Tripp" />

<meta name="date" content="2021-07-04" />

<title>Correlación</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Bioestadistica 2021</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Inicio</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/trippv/Bioestadistica2021">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Correlación</h1>
<h4 class="author">Miguel Tripp</h4>
<h4 class="date">2021-07-04</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-07-04
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>2021/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210412code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210412)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210412code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210412)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtrippvBioestadistica2021treef42c33c3016f56eb90af2514122705bc4916629atargetblankf42c33ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/trippv/Bioestadistica2021/tree/f42c33c3016f56eb90af2514122705bc4916629a" target="_blank">f42c33c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtrippvBioestadistica2021treef42c33c3016f56eb90af2514122705bc4916629atargetblankf42c33ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/trippv/Bioestadistica2021/tree/f42c33c3016f56eb90af2514122705bc4916629a" target="_blank">f42c33c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  Curso_Bioestadistica_MTripp_cuatriII.docx
    Untracked:  Curso_Bioestadistica_MTripp_cuatriII.pdf
    Untracked:  Diapositivas/
    Untracked:  Prueba_markdown.Rmd
    Untracked:  Prueba_markdown.pdf
    Untracked:  README.html
    Untracked:  Resources/
    Untracked:  Tarea_Tstudent.Rmd
    Untracked:  Tarea_Tstudent.docx
    Untracked:  Tarea_Tstudent.html
    Untracked:  Tarea_Tstudent.pdf
    Untracked:  analysis/images/
    Untracked:  code/tarea_macrograd.R
    Untracked:  data/CS_subset.csv
    Untracked:  data/Consumo_oxigeno_wide.csv
    Untracked:  data/Darwin_esp.csv
    Untracked:  data/Data_enzimas_Experimento1.txt
    Untracked:  data/Data_enzimas_Experimento2.txt
    Untracked:  data/Data_enzimas_Experimento3.txt
    Untracked:  data/Data_enzimas_Experimento4.txt
    Untracked:  data/DownloadFestival(No Outlier).dat
    Untracked:  data/Festival.csv
    Untracked:  data/LungCapData.txt
    Untracked:  data/LungCapDataEsp.csv
    Untracked:  data/RExam.dat
    Untracked:  data/Rexamendat.csv
    Untracked:  data/Tabla1_Muestreo.txt
    Untracked:  data/Transcriptome_Anotacion.csv
    Untracked:  data/Transcriptome_DGE.csv
    Untracked:  data/Vinogradov_2004_Titanic.tab
    Untracked:  data/Vinogradov_2004_Titanic.tab.csv
    Untracked:  data/data_tukey.txt
    Untracked:  data/datasets_Pokemon.csv
    Untracked:  data/datasets_Pokemon.xls
    Untracked:  data/exp_macrogard_growth.tab
    Untracked:  data/exp_macrogard_rna-dna.tab
    Untracked:  data/fertilizantes_luz.csv
    Untracked:  data/macrogard_crecimiento.csv
    Untracked:  data/penguins_size.csv
    Untracked:  data/pokemon_extended.csv
    Untracked:  output/Plot_all_penguins.pdf
    Untracked:  output/Plot_all_penguins.tiff
    Untracked:  output/graficos/

Unstaged changes:
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Clase10_correlacion.Rmd</code>) and HTML (<code>docs/Clase10_correlacion.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/Bioestadistica2021/392444fdd59cb7b5f5268d6186c85498d489dff8/docs/Clase10_correlacion.html" target="_blank">392444f</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-07-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/trippv/Bioestadistica2021/blob/4150ad45daf02450a26643c8c0bd79d9a3fa0db4/analysis/Clase10_correlacion.Rmd" target="_blank">4150ad4</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-07-04
</td>
<td>
Publish the initial files for myproject
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="generalidades" class="section level1" number="1">
<h1 number="1"><span class="header-section-number">1</span> Generalidades</h1>
<p>La correlación lineal y la regresión lineal simple son métodos estadísticos que estudian la relación lineal existente entre dos variables.</p>
<p>La correlación cuantifica como dos variables estan correlacionadas sin considerar dependencias. A nivel experimental, la correlación se suele emplear cuando ninguna de las variables se ha controlado, simplemente se han medido ambas y se desea saber si estan relacionadas.</p>
<p>Los <strong>coeficientes de correlación</strong> se pueden calcular mediante métodos paramétricos y no paramétricos. Un coeficiente paramétrico es el Coeficiente de Correlación de <strong>Pearson</strong>, que se usa para observaciones obtenidas sobre una escala de intervalos y esta sujeto a condiciones mas restrictivas que las alternativas no paramétricas. De estas, una de las mas ampliamente utilizadas es el Coeficiente de Correlación por Rangos de <strong>Spearman.</strong></p>
<p>Todos los coeficientes de corelación varían entre +1 y -1, siendo +1 una correlación positiva perfecta y -1 una correlación negativa perfecta:</p>
<p><img src="http://www.disfrutalasmatematicas.com/datos/images/correlation-examples.svg" /> Además del valor obtenido para el coeficiente de correlación, es necesario calcular su significancia. Solo si el <strong>valor p</strong> es significativo se puede aceptar que existe correlación, y esta será de la magnitud que indique el coeficiente.</p>
<p>Por muy cercano que sea el valor del coeficiente de correlación a +1 o −1, si no es significativo, se ha de interpretar que <strong>la correlación de ambas variables es 0</strong>, ya que el valor observado puede deberse a simple aleatoriedad.</p>
</div>
<div id="ejemplo1-calor-y-venta-de-helados" class="section level1" number="2">
<h1 number="2"><span class="header-section-number">2</span> Ejemplo1: Calor y venta de helados</h1>
<p><img src="https://images.unsplash.com/photo-1567206563064-6f60f40a2b57?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&amp;ixlib=rb-1.2.1&amp;auto=format&amp;fit=crop&amp;w=967&amp;q=80" /> Como primer ejemplo para el análisis de correlación, supongamos que una heladería realiza un seguimiento de la cantidad de helado que venden en relación con la temperatura de ese día. A continuación se muestran sus cifras de los últimos 12 días:</p>
<pre class="r"><code>library(tidyverse)
library(rstatix)
library(ggpubr)

temp &lt;- c(14.2, 16.4, 11.9, 15.2, 18.5, 22.1, 19.4, 25.1, 23.4, 18.1, 22.6, 17.2)

ventas &lt;- c(215, 325, 185, 332, 406, 522, 412, 614, 544, 421, 445, 408)

helados &lt;- data.frame(temp, ventas)

helados</code></pre>
<pre><code>   temp ventas
1  14.2    215
2  16.4    325
3  11.9    185
4  15.2    332
5  18.5    406
6  22.1    522
7  19.4    412
8  25.1    614
9  23.4    544
10 18.1    421
11 22.6    445
12 17.2    408</code></pre>
<div id="dispersión-de-los-datos" class="section level2" number="2.1">
<h2 number="2.1"><span class="header-section-number">2.1</span> Dispersión de los datos</h2>
<p>En primer lugar se representan las dos variables mediante un diagrama de dispersión. Este se puede construir simplemte con la función <strong>plot</strong> o con la función <strong>pairs()</strong></p>
<pre class="r"><code>plot(helados)</code></pre>
<p><img src="figure/Clase10_correlacion.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pairs(helados$temp ~ helados$ventas)</code></pre>
<p><img src="figure/Clase10_correlacion.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(helados, aes(x = temp, y = ventas))+
  geom_point()+
  labs(x = &quot;Temperatura (°C)&quot;, y = &quot;Ventas ($)&quot;, 
       title = &quot;ventas de helados&quot;)</code></pre>
<p><img src="figure/Clase10_correlacion.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>El diagrama de dispersión parecer indicar que existe una posible relación entre la temperatura del día con el número de ventas de helado.</p>
<p>Para poder elegir el coeficiente de correlación adecuando, se tiene que analizar el tipo de variable que se esta analizando y la distribución que presenta.</p>
</div>
<div id="análisis-de-normalidad" class="section level2" number="2.2">
<h2 number="2.2"><span class="header-section-number">2.2</span> Análisis de normalidad</h2>
<p>Utilizando métodos visuales con R base:</p>
<pre class="r"><code>par(mfrow = c(2,2))
hist(helados$temp, col = &quot;darkred&quot;)
qqnorm(helados$temp, col = &quot;darkred&quot;)
qqline(helados$temp)

hist(helados$ventas, col = &quot;blue&quot;)
qqnorm(helados$ventas, col = &quot;blue&quot;)
qqline(helados$ventas)</code></pre>
<p><img src="figure/Clase10_correlacion.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Realizando una prueba de Shapiro</p>
<pre class="r"><code>sapply(helados, shapiro_test)</code></pre>
<pre><code>          temp      ventas   
variable  &quot;X[[i]]&quot;  &quot;X[[i]]&quot; 
statistic 0.9759308 0.9615932
p.value   0.9620426 0.8063125</code></pre>
<p>El análisis gráfico y el contraste de normalidad muestran que para ambas variables se puede asumir normalidad. En caso contrario, se debe considerar el uso de alternativas como el calculo del coeficiente de <em>Spearman</em> o de <em>Kendall</em>.</p>
</div>
<div id="estimación-de-la-correlación" class="section level2" number="2.3">
<h2 number="2.3"><span class="header-section-number">2.3</span> Estimación de la correlación</h2>
<p>Para realizar la estimación de la correlación se puede utilizar la función <code>cor()</code>. Dentro del parámetro <em>method</em> podemos establecer cual coeficiente queremos utilizar (Pearson, Kendall o Spearman).</p>
<pre class="r"><code>cor(helados$ventas, helados$temp, method = &quot;pearson&quot;)</code></pre>
<pre><code>[1] 0.9575066</code></pre>
<pre class="r"><code>#La función tambien acepta data.frame o matrices, lo que nos arroja los coeficientes para todas las variables contenidas en la tabla

cor(helados)</code></pre>
<pre><code>            temp    ventas
temp   1.0000000 0.9575066
ventas 0.9575066 1.0000000</code></pre>
<p>La función nos arroja un coeficiente de correlación bastante alta (0.957). Sin embargo, por muy alto que pudiera ser, si no es significativa se ha de considerar inexistente. Para realizar la prueba de significancía de la correlación, se utiliza la función <code>cor.test()</code></p>
<pre class="r"><code>cor.test(helados$temp, helados$ventas, method = &quot;pearson&quot;, alternative = &quot;two.sided&quot;)</code></pre>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  helados$temp and helados$ventas
t = 10.499, df = 10, p-value = 1.016e-06
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.8515370 0.9883148
sample estimates:
      cor 
0.9575066 </code></pre>
<p>El coeficiente de correlación es altamente significativo por lo que podemos concluir que hay una correlación significativa entre la temperatura del día y la venta de helados en el dia (<span class="math inline">\(r = 0.957, valor p &lt; 0.0001\)</span>)</p>
</div>
<div id="importancia-de-la-visualización-en-la-correlación" class="section level2" number="2.4">
<h2 number="2.4"><span class="header-section-number">2.4</span> Importancia de la visualización en la correlación</h2>
<pre class="r"><code>library(pastecs)</code></pre>
<pre><code>
Attaching package: &#39;pastecs&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    first, last</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    extract</code></pre>
<pre class="r"><code>ascombe &lt;- datasets::anscombe

sapply(ascombe, stat.desc)</code></pre>
<pre><code>                     x1         x2         x3         x4         y1         y2
nbr.val      11.0000000 11.0000000 11.0000000 11.0000000 11.0000000 11.0000000
nbr.null      0.0000000  0.0000000  0.0000000  0.0000000  0.0000000  0.0000000
nbr.na        0.0000000  0.0000000  0.0000000  0.0000000  0.0000000  0.0000000
min           4.0000000  4.0000000  4.0000000  8.0000000  4.2600000  3.1000000
max          14.0000000 14.0000000 14.0000000 19.0000000 10.8400000  9.2600000
range        10.0000000 10.0000000 10.0000000 11.0000000  6.5800000  6.1600000
sum          99.0000000 99.0000000 99.0000000 99.0000000 82.5100000 82.5100000
median        9.0000000  9.0000000  9.0000000  8.0000000  7.5800000  8.1400000
mean          9.0000000  9.0000000  9.0000000  9.0000000  7.5009091  7.5009091
SE.mean       1.0000000  1.0000000  1.0000000  1.0000000  0.6125408  0.6125676
CI.mean.0.95  2.2281389  2.2281389  2.2281389  2.2281389  1.3648260  1.3648856
var          11.0000000 11.0000000 11.0000000 11.0000000  4.1272691  4.1276291
std.dev       3.3166248  3.3166248  3.3166248  3.3166248  2.0315681  2.0316567
coef.var      0.3685139  0.3685139  0.3685139  0.3685139  0.2708429  0.2708547
                     y3         y4
nbr.val      11.0000000 11.0000000
nbr.null      0.0000000  0.0000000
nbr.na        0.0000000  0.0000000
min           5.3900000  5.2500000
max          12.7400000 12.5000000
range         7.3500000  7.2500000
sum          82.5000000 82.5100000
median        7.1100000  7.0400000
mean          7.5000000  7.5009091
SE.mean       0.6121958  0.6122425
CI.mean.0.95  1.3640571  1.3641612
var           4.1226200  4.1232491
std.dev       2.0304236  2.0305785
coef.var      0.2707231  0.2707110</code></pre>
<p>Ahora estimamos los coeficientes de correlación para cada par de variables (x1, y1; x2, y2; x3, y3, etc)</p>
<pre class="r"><code>cor(ascombe)</code></pre>
<pre><code>           x1         x2         x3         x4         y1         y2         y3
x1  1.0000000  1.0000000  1.0000000 -0.5000000  0.8164205  0.8162365  0.8162867
x2  1.0000000  1.0000000  1.0000000 -0.5000000  0.8164205  0.8162365  0.8162867
x3  1.0000000  1.0000000  1.0000000 -0.5000000  0.8164205  0.8162365  0.8162867
x4 -0.5000000 -0.5000000 -0.5000000  1.0000000 -0.5290927 -0.7184365 -0.3446610
y1  0.8164205  0.8164205  0.8164205 -0.5290927  1.0000000  0.7500054  0.4687167
y2  0.8162365  0.8162365  0.8162365 -0.7184365  0.7500054  1.0000000  0.5879193
y3  0.8162867  0.8162867  0.8162867 -0.3446610  0.4687167  0.5879193  1.0000000
y4 -0.3140467 -0.3140467 -0.3140467  0.8165214 -0.4891162 -0.4780949 -0.1554718
           y4
x1 -0.3140467
x2 -0.3140467
x3 -0.3140467
x4  0.8165214
y1 -0.4891162
y2 -0.4780949
y3 -0.1554718
y4  1.0000000</code></pre>
<p>Ahora gráficamos cada uno de los pares</p>
<pre class="r"><code>par(mfrow = c(2,2))

plot(ascombe$x1, ascombe$y1, col = &quot;red&quot;, pch = 19)
plot(ascombe$x2, ascombe$y2, col = &quot;blue&quot;, pch = 19)
plot(ascombe$x3, ascombe$y3, col = &quot;darkorange&quot;, pch = 19)
plot(ascombe$x4, ascombe$y4, col = &quot;salmon&quot;, pch = 19)</code></pre>
<p><img src="figure/Clase10_correlacion.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
</div>
</div>
<div id="ejemplo-2-pokemon" class="section level1" number="3">
<h1 number="3"><span class="header-section-number">3</span> Ejemplo 2: Pokemon</h1>
<p><img src="https://images.unsplash.com/photo-1605979257913-1704eb7b6246?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&amp;ixlib=rb-1.2.1&amp;auto=format&amp;fit=crop&amp;w=1050&amp;q=80" /></p>
<p>Ahora vamos a análizar la correlación entre las distintas variables de la bases de datos de pokemon</p>
<pre class="r"><code>pokemon&lt;- readxl::read_excel(&quot;data/datasets_Pokemon.xls&quot;)

plot(pokemon)</code></pre>
<p><img src="figure/Clase10_correlacion.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Seleccionar el nombre y todas las variables numéricas</p>
<pre class="r"><code>pokemon_sub &lt;- pokemon %&gt;% 
  select(where(is.numeric), -Generation, -ID)

plot(pokemon_sub)</code></pre>
<p><img src="figure/Clase10_correlacion.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ivestigar variables correlacionadas</p>
<pre class="r"><code>cor(pokemon_sub)</code></pre>
<pre><code>            Total        HP    Attack   Defense    Sp_Atk    Sp_Def     Speed
Total   1.0000000 0.6187484 0.7362107 0.6127874 0.7472499 0.7176095 0.5759427
HP      0.6187484 1.0000000 0.4223860 0.2396223 0.3623799 0.3787181 0.1759521
Attack  0.7362107 0.4223860 1.0000000 0.4386871 0.3963618 0.2639896 0.3812397
Defense 0.6127874 0.2396223 0.4386871 1.0000000 0.2235486 0.5107466 0.0152266
Sp_Atk  0.7472499 0.3623799 0.3963618 0.2235486 1.0000000 0.5061214 0.4730179
Sp_Def  0.7176095 0.3787181 0.2639896 0.5107466 0.5061214 1.0000000 0.2591331
Speed   0.5759427 0.1759521 0.3812397 0.0152266 0.4730179 0.2591331 1.0000000</code></pre>
<p>Podemos usar la funcion <code>pairs.panel()</code> del paquete <strong>psych</strong> para visualziar los diagramas de disperción entre todas las variables asi como su valor de correlación.</p>
<pre class="r"><code>psych::pairs.panels(pokemon_sub, method = &quot;pearson&quot;, show.points = TRUE,stars = TRUE)</code></pre>
<p><img src="figure/Clase10_correlacion.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>El paquete Psych tambien contien la función multi.hist()la cual nos ayudaria a evaluar la distribución de los datos: Donde la linea roja corresponde a la distribución normal ajustada y la linea azul a la densidad observada.</p>
<pre class="r"><code>psych::multi.hist(pokemon_sub, density = TRUE,  dcol = c(&quot;blue&quot;, &quot;red&quot;))</code></pre>
<p><img src="figure/Clase10_correlacion.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>La función <code>corrplot()</code> del paquete <strong>corrplot</strong> recibe como argumento la matriz de correlaciones generada por la función cor() y genera diferentes tipos de heat maps mucho más visuales que la matriz numérica.</p>
<pre class="r"><code>library(corrplot)</code></pre>
<pre><code>corrplot 0.84 loaded</code></pre>
<pre class="r"><code>corrplot(corr = cor(x = pokemon_sub, method = &quot;pearson&quot;), method = &quot;number&quot;, type = &quot;upper&quot;)</code></pre>
<p><img src="figure/Clase10_correlacion.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>La función <code>ggpairs()</code> del paquete <strong>GGally</strong> basada en ggplot2 representa los diagramas de dispersión, el valor de la correlación e incluso interpreta el tipo de variable para que, en caso de ser categórica, representarla en forma de boxplot.</p>
<pre class="r"><code>library(GGally)</code></pre>
<pre><code>Registered S3 method overwritten by &#39;GGally&#39;:
  method from   
  +.gg   ggplot2</code></pre>
<pre class="r"><code>ggpairs(pokemon_sub, lower = list(continuous = &quot;smooth&quot;), 
        diag = list(continuous = &quot;bar&quot;), axisLabels = &quot;none&quot;)</code></pre>
<pre><code>Warning in check_and_set_ggpairs_defaults(&quot;diag&quot;, diag, continuous =
&quot;densityDiag&quot;, : Changing diag$continuous from &#39;bar&#39; to &#39;barDiag&#39;</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="figure/Clase10_correlacion.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<style>
div.blue { background-color:#e6f0ff; border-radius: 5px; padding: 20px;}
</style>
</div>
<div id="preguntas-frecuentes" class="section level1 blue" number="4">
<h1 number="4"><span class="header-section-number">4</span> Preguntas frecuentes</h1>
<p>Esta sección esta tomada del libro de Motuslky. <a href="https://books.google.com.mx/books?hl=en&amp;lr=&amp;id=ZLDWAgAAQBAJ&amp;oi=fnd&amp;pg=PP1&amp;dq=motulsky+intuitive+biostatistics+pdf&amp;ots=J1d8onYaXZ&amp;sig=c_dqfN7zgDoG5Tjy2-lSFll-Eew#v=onepage&amp;q=motulsky%20intuitive%20biostatistics%20pdf&amp;f=false">Intuitive Biostatistics</a></p>
<ul>
<li>¿Importa cual variable sea denominada <em>x</em> y cual es <em>y</em>?</li>
</ul>
<p>No. X y Y son completamente simétricas en los calculo de correlación. Sin embargo, esto no es cierto en el caso de la regresión (siguiente capitulo).</p>
<ul>
<li>¿<em>X</em> y <em>Y</em> tienen que ser medidas en la misma unidad para realizar los calculos de correlación?</li>
</ul>
<p>X y Y no tienen que ser medidas en la misma unidad pero pueden serlo.</p>
<ul>
<li>¿Por qué no hay una línea con el mejor ajuste en los gráficos de este capitulo?</li>
</ul>
<p>La correlación cuantifica la relación entre dos variables pero no ajusta un modelo a los datos, a diferencia de la regresión.</p>
<ul>
<li>¿Si se cambia el orden de <em>X</em> y <em>Y</em> también cambiará el valor de <span class="math inline">\(r\)</span>?</li>
</ul>
<p>No. X y Y son completamente simetricos en el calculo e interpretación del coeficiente de correlación.</p>
<ul>
<li>Puede el valor de <span class="math inline">\(r\)</span> ser expresado en porcentaje?</li>
</ul>
<p>No. El valor de r va de -1 a 1. No es una fracción por lo que no puede expresarse en porcentaje.</p>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19041)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] GGally_2.1.0    corrplot_0.84   pastecs_1.3.21  ggpubr_0.4.0   
 [5] rstatix_0.7.0   forcats_0.5.1   stringr_1.4.0   dplyr_1.0.5    
 [9] purrr_0.3.4     readr_1.4.0     tidyr_1.1.3     tibble_3.0.4   
[13] ggplot2_3.3.3   tidyverse_1.3.1 workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] nlme_3.1-152       fs_1.5.0           lubridate_1.7.10   RColorBrewer_1.1-2
 [5] httr_1.4.2         rprojroot_2.0.2    tools_4.0.5        backports_1.2.1   
 [9] utf8_1.2.1         R6_2.5.0           mgcv_1.8-33        DBI_1.1.0         
[13] colorspace_2.0-0   withr_2.4.2        tidyselect_1.1.1   mnormt_2.0.2      
[17] curl_4.3           compiler_4.0.5     git2r_0.27.1       cli_2.5.0         
[21] rvest_1.0.0        xml2_1.3.2         labeling_0.4.2     scales_1.1.1      
[25] psych_2.0.12       digest_0.6.27      foreign_0.8-81     rmarkdown_2.6     
[29] rio_0.5.16         pkgconfig_2.0.3    htmltools_0.5.1.1  dbplyr_2.1.1      
[33] rlang_0.4.11       readxl_1.3.1       rstudioapi_0.13    farver_2.0.3      
[37] generics_0.1.0     jsonlite_1.7.2     zip_2.1.1          car_3.0-10        
[41] magrittr_2.0.1     Matrix_1.3-2       Rcpp_1.0.5         munsell_0.5.0     
[45] fansi_0.4.2        abind_1.4-5        lifecycle_1.0.0    stringi_1.5.3     
[49] whisker_0.4        yaml_2.2.1         carData_3.0-4      plyr_1.8.6        
[53] grid_4.0.5         parallel_4.0.5     promises_1.1.1     crayon_1.4.1      
[57] lattice_0.20-41    splines_4.0.5      haven_2.3.1        hms_1.0.0         
[61] tmvnsim_1.0-2      knitr_1.30         ps_1.5.0           pillar_1.6.0      
[65] boot_1.3-27        ggsignif_0.6.0     reprex_2.0.0       glue_1.4.2        
[69] evaluate_0.14      data.table_1.13.6  modelr_0.1.8       vctrs_0.3.8       
[73] httpuv_1.5.4       cellranger_1.1.0   gtable_0.3.0       reshape_0.8.8     
[77] assertthat_0.2.1   xfun_0.23          openxlsx_4.2.3     broom_0.7.6       
[81] later_1.1.0.1      ellipsis_0.3.1    </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = false;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
