<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Miguel Tripp" />

<meta name="date" content="2021-04-25" />

<title>Clase3 Graficos Base</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Bioestadistica 2021</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/trippv/Bioestadistica2021">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Clase3 Graficos Base</h1>
<h4 class="author">Miguel Tripp</h4>
<h4 class="date">2021-04-25</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-06-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>2021/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210412code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210412)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210412code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210412)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtrippvBioestadistica2021tree624db7cbaae7bec3f471d3e90e7870927f224bd3targetblank624db7ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/trippv/Bioestadistica2021/tree/624db7cbaae7bec3f471d3e90e7870927f224bd3" target="_blank">624db7c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtrippvBioestadistica2021tree624db7cbaae7bec3f471d3e90e7870927f224bd3targetblank624db7ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/trippv/Bioestadistica2021/tree/624db7cbaae7bec3f471d3e90e7870927f224bd3" target="_blank">624db7c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  Curso_Bioestadistica_MTripp_cuatriII.docx
    Untracked:  Curso_Bioestadistica_MTripp_cuatriII.pdf
    Untracked:  Diapositivas/
    Untracked:  README.html
    Untracked:  Resources/
    Untracked:  analysis/images/
    Untracked:  data/CS_subset.csv
    Untracked:  data/Consumo_oxigeno_wide.csv
    Untracked:  data/Data_enzimas_Experimento1.txt
    Untracked:  data/Data_enzimas_Experimento2.txt
    Untracked:  data/Data_enzimas_Experimento3.txt
    Untracked:  data/Data_enzimas_Experimento4.txt
    Untracked:  data/DownloadFestival(No Outlier).dat
    Untracked:  data/Festival.csv
    Untracked:  data/LungCapData.txt
    Untracked:  data/LungCapDataEsp.csv
    Untracked:  data/RExam.dat
    Untracked:  data/Rexamendat.csv
    Untracked:  data/Tabla1_Muestreo.txt
    Untracked:  data/Transcriptome_Anotacion.csv
    Untracked:  data/Transcriptome_DGE.csv
    Untracked:  data/datasets_Pokemon.csv
    Untracked:  data/datasets_Pokemon.xls
    Untracked:  data/exp_macrogard_growth.tab
    Untracked:  data/exp_macrogard_rna-dna.tab
    Untracked:  data/macrogard_crecimiento.csv
    Untracked:  data/penguins_size.csv
    Untracked:  data/pokemon_extended.csv
    Untracked:  output/Plot_all_penguins.pdf
    Untracked:  output/Plot_all_penguins.tiff
    Untracked:  output/graficos/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Clase3_GraficosBase.Rmd</code>) and HTML (<code>docs/Clase3_GraficosBase.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/Bioestadistica2021/fbd7857f5a9cf55d6546a2247dd2de761889b4af/docs/Clase3_GraficosBase.html" target="_blank">fbd7857</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-06-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/Bioestadistica2021/87a646f0f99382cc3dd71a37665b3b42c2977675/docs/Clase3_GraficosBase.html" target="_blank">87a646f</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-06-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/Bioestadistica2021/47a71474d287b7e473979ca0ec49cc4e7b28cc37/docs/Clase3_GraficosBase.html" target="_blank">47a7147</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-05-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/Bioestadistica2021/95c661f0de4c8358e2c8122680fb73ea8d8612a7/docs/Clase3_GraficosBase.html" target="_blank">95c661f</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-05-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/Bioestadistica2021/34660dee6ce175fda7ed3d94a4ab1a94a53de330/docs/Clase3_GraficosBase.html" target="_blank">34660de</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-05-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/Bioestadistica2021/da04854bc9fd56374352c0be63851caeecf23d51/docs/Clase3_GraficosBase.html" target="_blank">da04854</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-05-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/Bioestadistica2021/3321935893046310778d4aa798117c5d525f8bdb/docs/Clase3_GraficosBase.html" target="_blank">3321935</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-05-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/Bioestadistica2021/3d129029da9d50a930facaf1e77511a5129c79ff/docs/Clase3_GraficosBase.html" target="_blank">3d12902</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-05-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/Bioestadistica2021/20b3aabb9d7f66c67502a9843570429fff917f3c/docs/Clase3_GraficosBase.html" target="_blank">20b3aab</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-05-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/Bioestadistica2021/f20dcb1ae45a1c0a5d5e73ad53b434830114d82e/docs/Clase3_GraficosBase.html" target="_blank">f20dcb1</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-05-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/Bioestadistica2021/b1ca7f341b4a6dbf2c3c4b96ed215822e6927c2c/docs/Clase3_GraficosBase.html" target="_blank">b1ca7f3</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-05-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/Bioestadistica2021/b3002cc4120ff646c9651b902e77971c28b56ad4/docs/Clase3_GraficosBase.html" target="_blank">b3002cc</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-05-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/Bioestadistica2021/2c60db3bc18304f98380b5c49229c69b36269a21/docs/Clase3_GraficosBase.html" target="_blank">2c60db3</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-05-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/Bioestadistica2021/65fa0eede96c6410437f2cce9e928104694aa749/docs/Clase3_GraficosBase.html" target="_blank">65fa0ee</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-05-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/Bioestadistica2021/b3c3d89f7e3635396cbe8ee82f281f84bc3caf14/docs/Clase3_GraficosBase.html" target="_blank">b3c3d89</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/Bioestadistica2021/195c2e9056acf18cfae696f9ea9ef73acf080751/docs/Clase3_GraficosBase.html" target="_blank">195c2e9</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-04-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/Bioestadistica2021/e322dacb825a42b4675e367cc54615205445dc6d/docs/Clase3_GraficosBase.html" target="_blank">e322dac</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-04-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/trippv/Bioestadistica2021/blob/7d0111c0e1a10dab8ceab4a801d08db523858c6d/analysis/Clase3_GraficosBase.Rmd" target="_blank">7d0111c</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-04-25
</td>
<td>
Publish the initial files for myproject
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Uno de los aspectos mas poderosos de R es su capacidad de producir una gran variedad de gráficos los cuales pueden ser reproducibles, modificables y publicables con solo unos poco comandos.</p>
<p>R puede producir gráficos utilizando tres paquetes principales: ggplot, lattice y gráficos base. En esta sección nos enfocaremos a los gráficos base.</p>
<p>Para estos ejercicios, utilizaremos la base de datos de pingünos del Archipielago de Palmer la cual se encuentra disponible en la carpeta <code>data</code></p>
<pre class="r"><code>penguins &lt;- read.table(&quot;data/penguins_size.csv&quot;, sep = &quot;,&quot;, na.strings = &quot;NA&quot;, header = TRUE, as.is = FALSE)
penguins &lt;- penguins[complete.cases(penguins), ]</code></pre>
<p>El archivo <code>penguins_size.csv</code> contiene las variables:</p>
<ul>
<li><code>species</code>: Especies de pingüino (Chnstrap, Adelie o Gentoo)</li>
<li><code>culmen_length_mm</code>: longitud del culmen (mm)</li>
<li><code>culmen_depth_mm</code>: profundidad del culmen</li>
<li><code>flipper_length_mm</code>: longitud de la aleta (mm)</li>
<li><code>body_mass_g</code>: masa corporal (g)</li>
<li><code>island</code>: nombre de la isla (Dream, Torgersen o Biscoe)</li>
<li><code>sex</code>: sexo</li>
</ul>
<p><strong>Los pingüinos:</strong></p>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/allisonhorst/palmerpenguins/master/man/figures/lter_penguins.png" alt="@AllisonHorst" width="100%" />
<p class="caption">
<span class="citation">@AllisonHorst</span>
</p>
</div>
<div class="figure" style="text-align: center">
<img src="https://raw.githubusercontent.com/allisonhorst/palmerpenguins/master/man/figures/culmen_depth.png" alt="@AllisonHorst" width="100%" />
<p class="caption">
<span class="citation">@AllisonHorst</span>
</p>
</div>
<div id="histograma-hist" class="section level2">
<h2>1: Histograma <code>hist()</code></h2>
<p>Para crear un histograma usamos la función <code>hist()</code>, que siempre nos pide como argumento <em>x</em> un vector numérico. El resto de los argumentos de esta función son opcionales. Si damos un vector no numérico, se nos devolverá un error.</p>
<p>Para obtener un hitograma de frecuencias del peso corporal <em>body_mass_g</em> de todos los pingüinos, se ejecuta el siguiente comando:</p>
<pre class="r"><code>hist(x = penguins$body_mass_g)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Para la longitud del culmen <em>culmen_lengh_mm</em>:</p>
<pre class="r"><code>hist(x = penguins$culmen_length_mm)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Vamos a cambiar el título del gráfico con el argumento <code>main</code>, y el nombre de los ejes x y Y con <code>xlab</code> y <code>ylab</code>, respectivamente.</p>
<pre class="r"><code>hist(x = penguins$culmen_length_mm,
     
     #colora el titulo principal
     main = &quot;histograma de longitud del culmen (mm)&quot;,
     
     #etiqueta del eje x
     xlab = &quot;longitud (mm)&quot;,
     
     #etiqueta del eje y
     ylab = &quot;Frecuencia&quot;)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>La funcion hist auntomaticamente selecciona el número de intervales (“<em>breaks</em>”) mas adecuado. Pero es posible definirlos manualmente con el parametro “breaks”</p>
<pre class="r"><code>hist(x = penguins$culmen_length_mm,
     breaks = 20)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist(x = penguins$culmen_length_mm,
     breaks = 30)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Probemos cambiando el color de las barras del histograma agregando el argumento col. Este argumento acepta nombres de colores genéricos en inglés como “red”, “blue” o “purple”; y también acepta colores hexadecimales, como “#00FFFF”, “#08001a” o “#1c48b5”.</p>
<pre class="r"><code>hist(x = penguins$culmen_length_mm,
     main = &quot;histograma de longitud total (mm)&quot;,
     xlab = &quot;longitud (mm)&quot;,
     ylab = &quot;Frecuencia&quot;,
     col = &quot;lightblue&quot;)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Y finalmente podemos añadir dos histogramas en una sola grafica. Para esto, baso a aplicar lo que hemos visto hasta ahora para generar un subset con los datos de longitud del culmen <em>culmen_length_mm</em> de los machos y otro de las hembras</p>
<p>Posteriormente, se llama la función <code>hist()</code> para cada subset, pero al segundo se le debe añadir el parametro: <code>add = TRUE</code></p>
<pre class="r"><code>#generar el subset de dato de los machos
male_peng &lt;- penguins[penguins$sex == &quot;MALE&quot;, &quot;culmen_length_mm&quot;]

#subset de datos de hembrsa
female_peng &lt;- penguins[penguins$sex == &quot;FEMALE&quot;,&quot;culmen_length_mm&quot; ]


#generar el primer histograma
hist(male_peng, breaks = 25, col = &quot;lightblue&quot;,
     main = &quot;histograma de frecuencias por sexo&quot;,
     ylab = &quot;Frecuencia&quot;,
     xlab = &quot;Longitud (mm)&quot;)

hist(female_peng, breaks = 25, col = &quot;salmon&quot;, add = TRUE)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="función-plot" class="section level2">
<h2>2:Función <code>plot()</code></h2>
<p>Esta función tiene un comportamiento especial, pues dependiendo del tipo de dato que le demos como argumento, generará diferentes tipos de gráfica. Además, para cada tipo de gráfico, podremos ajustar diferentes parámetros que controlan su aspecto, dentro de esta misma función.</p>
<p>Dependiendo del tipo de dato que demos a x y y será el gráfico que obtendremos, de acuerdo a las siguientes reglas:</p>
<table>
<thead>
<tr class="header">
<th>x</th>
<th>y</th>
<th>Gráfico</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Continuo</td>
<td>Continuo</td>
<td>Diagrama de dispersión (Scatterplot)</td>
</tr>
<tr class="even">
<td>Continuo</td>
<td>Discreto</td>
<td>Diagrama de dispersión, y coercionada a numérica</td>
</tr>
<tr class="odd">
<td>Continuo</td>
<td>Ninguno</td>
<td>Diagrama de dispersión, por número de renglón</td>
</tr>
<tr class="even">
<td>Discreto</td>
<td>Continuo</td>
<td>Diagrama de caja (Box plot)</td>
</tr>
<tr class="odd">
<td>Discreto</td>
<td>Discreto</td>
<td>Gráfico de mosaico (Diagrama de Kinneman)</td>
</tr>
<tr class="even">
<td>Discreto</td>
<td>Ninguno</td>
<td>Gráfica de barras</td>
</tr>
<tr class="odd">
<td>Ninguno</td>
<td>Cualquiera</td>
<td>Error</td>
</tr>
</tbody>
</table>
<p>por ejemplo, si graficamos solamente la longitud del culmen (continuo + Ninguno):</p>
<pre class="r"><code>plot(penguins$culmen_length_mm)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>pero si graficamos longitud del culmen por la masa corporal (continuo + continuo)</p>
<pre class="r"><code>plot(x = penguins$culmen_length_mm, y = penguins$body_mass_g)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>ahora si graficamos Longitud total x Sitio (continuo x discreto)</p>
<pre class="r"><code>plot(penguins$culmen_length_mm, penguins$species)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>si graficamos la especies (discreto x niunguno)</p>
<pre class="r"><code>plot(penguins$species)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ahora si graficamos isla x sexo (discreto x discreto)</p>
<pre class="r"><code>plot(penguins$island, penguins$sex)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>ahora si graficamos isla y longitud total (discreto x continuo)</p>
<pre class="r"><code>plot(penguins$island,  penguins$culmen_length_mm)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>La función plot es particularmente util para graficar series de tiempo. Para visualizar esto, vamos a generar un vector con una serie de 31 años, desde 1990 hasta el 2020 usando la función <code>seq()</code> y <code>rnorm()</code></p>
<pre class="r"><code>year &lt;- seq(1990,2020,1)
value &lt;- rnorm(31)

plot(year, value)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Para controlar el tipo de grafico dentre de esta función, usamos el parametro `type" donde puede ser:</p>
<ul>
<li>“l” = linea</li>
<li>“o” = puntos</li>
<li>“b” = ambos</li>
</ul>
<pre class="r"><code>plot(year, value, type = &quot;l&quot;)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(year, value, type =&quot;b&quot;)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Al igual que con <code>hist()</code> es posible cambiar el color de una grafica, asi como los nombres usando los parametros <code>main</code>, <code>col</code> y <code>xlab</code>, <code>ylab</code></p>
<pre class="r"><code>plot(year, value, type=&quot;l&quot;, col =&quot;red&quot;, main = &quot;gráfico anual&quot;, xlab = &quot;Año&quot;, ylab = &quot;anomalia&quot;)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="boxplots-boxplot" class="section level2">
<h2>3:Boxplots <code>boxplot()</code></h2>
<p>Los diagrams de caja, o boxplots, son gráficos que muestra la distribución de una variable usando cuartiles, de modo que de manera visual podemos inferir algunas cosas sobre su dispersión, ubicación y simetría.</p>
<p>Una gráfica de este tipo dibuja un rectángulo cruzado por una línea recta horizontal. Esta linea recta representa la <strong>mediana</strong>, el segundo cuartil, su base representa el pimer cuartil y su parte superior el tercer cuartil. Al rango entre el primer y tercer cuartil se le conoce como <strong>intercuartílico</strong>.</p>
<p>EN la sección anterior vimos como generar un boxplot usando la funcion <code>plot()</code>. Una alternativa es utilizar la función <code>boxplot()</code>.</p>
<p>En la segunda manera necesitamos dar dos argumentos:</p>
<ul>
<li>formula: Para esta función las fórmulas tienen el formato y ~ x, donde x es el nombre de la variable continua a graficar, y la x es la variable que usaremos como agrupación.</li>
<li>data: Es el data frame del que serántomadas las variable</li>
</ul>
<p>Por ejemplo, para generar diagramas de la longitud total por sitio de colecta, se ejecuta el comando:</p>
<pre class="r"><code>boxplot(culmen_length_mm ~ island, data = penguins)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Esta opción nos permite generar graficos con interacción. Por ejemplo, si quieremos graficar la longitud total en función del sitio de colecta y del sexo, ejecutamos:</p>
<pre class="r"><code>boxplot(culmen_length_mm ~ sex * island, data = penguins)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-22-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Para facilitar la lectura de la gráfica, se le puede asignar un color difente a cada grupo</p>
<pre class="r"><code>boxplot(culmen_length_mm ~ sex * island, data = penguins, col = c(&quot;tomato&quot;, &quot;lightblue&quot;))</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-23-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="edición" class="section level2">
<h2>4:Edición</h2>
<p>Asignar un color de acuerdo a una variable con el parametro <code>col</code></p>
<pre class="r"><code>plot(x = penguins$culmen_length_mm, 
     y = penguins$body_mass_g, 
     col = penguins$sex)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Cambiar el tamaño del simbolo o forma del simbolo:</p>
<pre class="r"><code>plot(x = penguins$culmen_length_mm, 
     y = penguins$body_mass_g, 
     col = penguins$sex,
     pch = 16,
     cex = 2)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>En donde <code>pch()</code> indica el simbolo, mientras que <code>cex()</code>el tamaño.</p>
<p>Aqui encontraras una lista con los valores de cada simbolo: <img src="http://www.sthda.com/sthda/RDoc/figure/graphs/r-plot-pch-symbols-points-in-r.png"></p>
<p>Para los simbolos de 21 a 25, es necesario especificar el color del borde (<code>col=</code>) asi como del relleno (<code>bg=</code>)</p>
<p>CUando usamos <code>plot()</code> es posible unir dos series diferentes. Para esto, primero se gráfica la primera serie, que en este caso es “value” y posteriormente se añada la segunda serie usando la funciones <code>lines()</code>. Por ejemplo, generamos un segundo vector con 31 datos:</p>
<pre class="r"><code>value2 &lt;- rnorm(31)</code></pre>
<p>Ahora podemos unir ambos vectores en la misma gráfica de la siguiente forma:</p>
<pre class="r"><code>plot(year, value, type = &quot;l&quot;, col =&quot;tomato&quot;, lwd=2, ylim = c(-5,5))
lines(year, value2, col = &quot;lightblue&quot;, lwd = 2, lty= &quot;dashed&quot;)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Entre los parametros que se pueden modificar tenemos:</p>
<ul>
<li><code>lwd</code> = grosor de la linea</li>
<li><code>lty</code> = tipo de linea (“dashed”, “solid”, “dotted” y “longdash”)</li>
<li><code>ylim</code> = limites del eje Y. Este es en forma de vector numerico</li>
</ul>
<p>Para añadir una leyenda, se usa la función <code>legend()</code> despues del gráfico.</p>
<pre class="r"><code>plot(year, value, type = &quot;l&quot;, col =&quot;tomato&quot;, lwd=2, ylim = c(-5,5))
lines(year, value2, col = &quot;lightblue&quot;, lwd = 2, lty= &quot;dashed&quot;)
legend(&quot;topleft&quot;, legend = c(&quot;value1&quot;, &quot;value2&quot;),
       col=c(&quot;tomato&quot;, &quot;lightblue&quot;),
       lty = c(&quot;solid&quot;, &quot;dashed&quot;))</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Para unir dos o mas gráficos es posible ajustar el área de gráfico con la funcion <code>par()</code></p>
<pre class="r"><code>par(mfrow=c(1,2)) #numero de filas, numero de columnas

plot(year, value, type = &quot;l&quot;, col =&quot;tomato&quot;, lwd=2, ylim = c(-5,5))

plot(year, value2, type = &quot;l&quot;, col =&quot;lightblue&quot;, lwd=2, ylim = c(-5,5))</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dev.off() #limpia el area de gráfico</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>par(mfrow=c(2,1)) #numero de filas, numero de columnas

plot(year, value, type = &quot;l&quot;, col =&quot;tomato&quot;, lwd=2, ylim = c(-5,5))

plot(year, value2, type = &quot;l&quot;, col =&quot;lightblue&quot;, lwd=2, ylim = c(-5,5))</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<p>Finalmente, es posible usar la función <code>mtext()</code> para agregar titulo a una serie de gráficas.</p>
<pre class="r"><code>par(mfrow=c(1,3), oma = c(0, 0, 2, 0)) #numero de filas, numero de columnas

# Grafico de dispersión
plot(x = penguins$culmen_length_mm, 
     y = penguins$body_mass_g, 
     col = penguins$sex,
     type = &quot;p&quot;,
     main = &quot;Relacion entre longitud del culmen y masa corporal&quot;,
     pch = 15,
     ylab = &quot;Masa corporal (g)&quot;,
     xlab = &quot;Longitud del culmen (mm)&quot;)


# Boxplot

boxplot(culmen_length_mm ~ island, data = penguins, main = &quot;Longitud del culmen por sitio&quot;)

# Histograma
hist(penguins$culmen_length_mm, main=&quot;Histograma Logitud del culmen (mm)&quot;)

mtext(&quot;Datos de longitud del culmen&quot;, outer = TRUE)</code></pre>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dev.off() #limpia el area de gráfico</code></pre>
<pre><code>null device 
          1 </code></pre>
<p>Observa que para que se despliegue el texto en la parte superior, es necesario ajustar el espacio del gráfico, es decir, incrementar el margen</p>
<p>Los margenes de una figura se ajustan con los parametros <code>mar()</code>, para los margenes internos, y <code>omar()</code>para los margenes externos, tal como se muestra en la siguiente figura:</p>
<p><img src="https://www.r-graph-gallery.com/74-margin-and-oma-cheatsheet_files/figure-html/thecode-1.png"></p>
<p>Para ambos parametros, se asignan cuatro valores para indicar el espacio abajo, izquierda, arriba, derecha.</p>
<hr />
</div>
<div id="ejercicio" class="section level2">
<h2>Ejercicio:</h2>
<blockquote>
<p><strong>Ejercicio:</strong> Usando todos tus conocimientos adquiridos hasta este punto, recrea esta gráfica:</p>
</blockquote>
<p><img src="figure/Clase3_GraficosBase.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>null device 
          1 </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19041)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5        whisker_0.4       knitr_1.30        magrittr_2.0.1   
 [5] R6_2.5.0          rlang_0.4.11      fansi_0.4.2       highr_0.8        
 [9] stringr_1.4.0     tools_4.0.5       xfun_0.20         utf8_1.2.1       
[13] git2r_0.27.1      htmltools_0.5.1.1 ellipsis_0.3.1    rprojroot_2.0.2  
[17] yaml_2.2.1        digest_0.6.27     tibble_3.0.4      lifecycle_1.0.0  
[21] crayon_1.4.1      later_1.1.0.1     vctrs_0.3.8       promises_1.1.1   
[25] fs_1.5.0          glue_1.4.2        evaluate_0.14     rmarkdown_2.6    
[29] stringi_1.5.3     compiler_4.0.5    pillar_1.6.0      httpuv_1.5.4     
[33] pkgconfig_2.0.3  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
