<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Miguel Tripp" />

<meta name="date" content="2021-07-07" />

<title>Regresión lineal multiple</title>

<script src="site_libs/header-attrs-2.9/header-attrs.js"></script>
<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.17/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Bioestadistica en R</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Inicio</a>
</li>
<li>
  <a href="about.html">Información del curso</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Modulos
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Básicos</li>
    <li>
      <a href="Clase1_Basicos.html">1. Introducción y comandos básicos</a>
    </li>
    <li>
      <a href="Clase2_Data_Frames.html">2. Data frames, matrices y otros objetovs</a>
    </li>
    <li>
      <a href="Clase3_GraficosBase.html">3. Gráficos base</a>
    </li>
    <li>
      <a href="CLase4_Dplyr.html">4. Dplyr y Tidyverse</a>
    </li>
    <li>
      <a href="Clase5_ggplot.html">5. Gráficos con ggplot</a>
    </li>
    <li class="dropdown-header">inferencia estadística</li>
    <li>
      <a href="Clase6_Supuestos.html">6. Supuestos normalidad y varianza</a>
    </li>
    <li>
      <a href="Clase7_tstudent.html">7. T de Student</a>
    </li>
    <li>
      <a href="Clase8_anova.html">8. ANOVA</a>
    </li>
    <li>
      <a href="Clase9_anova2vias.html">9. ANOVA 2 vias</a>
    </li>
    <li>
      <a href="Clase10_correlacion.html">10. Correlación</a>
    </li>
    <li>
      <a href="Clase11_regresion.html">11. Regresión líneal simple</a>
    </li>
    <li>
      <a href="Clase12_regresionMulti.html">12. Regresión líneal multiple</a>
    </li>
    <li>
      <a href="Clase13_noParam.html">13. No paramétricos</a>
    </li>
    <li>
      <a href="Clase14_PCA.html">14. PCA</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/trippv/BioestadisticaR">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Regresión lineal multiple</h1>
<h4 class="author">Miguel Tripp</h4>
<h4 class="date">2021-07-07</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-08-01
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>2021/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210412code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210412)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210412code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210412)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtrippvBioestadisticaRtree76287204bfc7f4af77792d7636f12d797930d2abtargetblank7628720a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/trippv/BioestadisticaR/tree/76287204bfc7f4af77792d7636f12d797930d2ab" target="_blank">7628720</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtrippvBioestadisticaRtree76287204bfc7f4af77792d7636f12d797930d2abtargetblank7628720a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/trippv/BioestadisticaR/tree/76287204bfc7f4af77792d7636f12d797930d2ab" target="_blank">7628720</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/hero-image.html
    Ignored:    analysis/poke_logo.png

Untracked files:
    Untracked:  Curso_Bioestadistica_MTripp_cuatriII.docx
    Untracked:  Curso_Bioestadistica_MTripp_cuatriII.pdf
    Untracked:  Diapositivas/
    Untracked:  Prueba_markdown.Rmd
    Untracked:  Prueba_markdown.pdf
    Untracked:  README.html
    Untracked:  Resources/
    Untracked:  Tarea_Tstudent.Rmd
    Untracked:  Tarea_Tstudent.docx
    Untracked:  Tarea_Tstudent.html
    Untracked:  Tarea_Tstudent.pdf
    Untracked:  analysis/Clase13_noParam.Rmd
    Untracked:  analysis/hero_backgroud.jpg
    Untracked:  analysis/images/
    Untracked:  analysis/style.css
    Untracked:  analysis/test.Rmd
    Untracked:  code/tarea_macrograd.R
    Untracked:  data/CS_subset.csv
    Untracked:  data/Consumo_oxigeno_wide.csv
    Untracked:  data/Darwin_esp.csv
    Untracked:  data/Data_enzimas_Experimento1.txt
    Untracked:  data/Data_enzimas_Experimento2.txt
    Untracked:  data/Data_enzimas_Experimento3.txt
    Untracked:  data/Data_enzimas_Experimento4.txt
    Untracked:  data/DownloadFestival(No Outlier).dat
    Untracked:  data/Festival.csv
    Untracked:  data/Hful_metabolitos_ver2.csv
    Untracked:  data/Longitud_noParam.csv
    Untracked:  data/LungCapData.txt
    Untracked:  data/LungCapDataEsp.csv
    Untracked:  data/PalmerPenguins.csv
    Untracked:  data/Pokemon_tabla.csv
    Untracked:  data/Pokemon_tabla.xls
    Untracked:  data/RExam.dat
    Untracked:  data/Rexamendat.csv
    Untracked:  data/Tabla1_Muestreo.txt
    Untracked:  data/Transcriptome_Anotacion.csv
    Untracked:  data/Transcriptome_DGE.csv
    Untracked:  data/Vinogradov_2004_Titanic.tab
    Untracked:  data/Vinogradov_2004_Titanic.tab.csv
    Untracked:  data/data_tukey.txt
    Untracked:  data/exp_macrogard_growth.tab
    Untracked:  data/exp_macrogard_rna-dna.tab
    Untracked:  data/fertilizantes_luz.csv
    Untracked:  data/gatos_sueno.csv
    Untracked:  data/macrogard_crecimiento.csv
    Untracked:  data/pokemon_extended.csv
    Untracked:  output/Plot_all_penguins.pdf
    Untracked:  output/Plot_all_penguins.tiff
    Untracked:  output/graficos/

Unstaged changes:
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Clase12_regresionMulti.Rmd</code>) and HTML (<code>docs/Clase12_regresionMulti.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/BioestadisticaR/5aafed258f3a7685091e5c6b3ac2cc2611cbbcb4/docs/Clase12_regresionMulti.html" target="_blank">5aafed2</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-08-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/BioestadisticaR/2bc601affbc9f3bcec07e7fb128cea0084fb32ce/docs/Clase12_regresionMulti.html" target="_blank">2bc601a</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-07-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/trippv/BioestadisticaR/blob/b4e67dc1256ad5367ecc659603bbd5df3c597ea7/analysis/Clase12_regresionMulti.Rmd" target="_blank">b4e67dc</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-07-31
</td>
<td>
workflowr::wflow_publish(c(“analysis/index.Rmd”, “analysis/about.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/BioestadisticaR/03db7ffb0734a19d5fa2e2a28ffc934c8bc1f20a/docs/Clase12_regresionMulti.html" target="_blank">03db7ff</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-07-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/BioestadisticaR/fcb90476bed084093e76c71117c1c0b51a8d9bd1/docs/Clase12_regresionMulti.html" target="_blank">fcb9047</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-07-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/BioestadisticaR/2adc7a9da7a482c216c0aaec3f3eca49c734bbf5/docs/Clase12_regresionMulti.html" target="_blank">2adc7a9</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-07-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/BioestadisticaR/9d094205eea0b6e1dce21121975c2a3463bd21d4/docs/Clase12_regresionMulti.html" target="_blank">9d09420</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-07-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/trippv/BioestadisticaR/0f7eb2db0a3abc3d1d590d8d25368fe160a28a0b/docs/Clase12_regresionMulti.html" target="_blank">0f7eb2d</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-07-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/trippv/BioestadisticaR/blob/9e1edceb5867ba1d086e3478b340870da686da9f/analysis/Clase12_regresionMulti.Rmd" target="_blank">9e1edce</a>
</td>
<td>
Miguel Tripp
</td>
<td>
2021-07-12
</td>
<td>
Publish the initial files for myproject
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="generalidades" class="section level1" number="1">
<h1 number="1"><span class="header-section-number">1</span> Generalidades</h1>
<p>La regresión lineal simple asume que hay una sola variable predictora en la que estamos interesados. Sin embargo existen muchas situaciones en la que queremos predicir un valor a partir de multiples variables predictora, por lo que necesitamos de una extensión de la reresión lineal.</p>
<p>La regresión lineal múltiple permite generar un modelo lineal en el que el valor de la variable dependiente o respuesta (<span class="math inline">\(Y\)</span>) se determina a partir de un conjunto de variables independientes llamadas predictores (<span class="math inline">\(X_1\)</span>, <span class="math inline">\(X_2\)</span>, <span class="math inline">\(X_3\)</span>…).</p>
<p>Conceptualmente, los modelos lineales multiples son muy sencillos. Todo lo que necesitamos hacer es añadir mas terminos a nuestra ecuación de regresión, de manera que pasamos de una ecuación simple como esta:</p>
<p><span class="math inline">\(Y=\beta_0 + \beta_1 X_i + \epsilon\)</span></p>
<p>a esta:</p>
<p><span class="math inline">\(Y=\beta_0 + \beta_1 X_{1i} + \beta_2 X_{2i} + ... + \beta_n X_{ni} + \epsilon\)</span></p>
<p>Al igual que en nuestro modelo lineal sencillo, <span class="math inline">\(\epsilon_i\)</span> corresponde a los residuales asociados a la <span class="math inline">\(i_{esima}\)</span> observación (<span class="math inline">\(\epsilon_i = Y_i - \hat{Y}_i\)</span>). Pero en este modelo ahora tenemos tres coeficientes que tienen que ser estimados: el intercepto (<span class="math inline">\(b_0\)</span>), la pendiente para la variable i (<span class="math inline">\(b_1\)</span>) y la pendiente para la vairable 2 (<span class="math inline">\(b_2\)</span>).</p>
<p>A pesar de que ahora tenemos mas coeficientes que necesitan ser estimados, la idea básica de la estimación sigue siendo la misma: los coeficientes estimados son aquellos que minimizan la suma de cuadrado de los residuales.</p>
</div>
<div id="ejemplo-sencillo-gatos-y-horas-de-sueño" class="section level1" number="2">
<h1 number="2"><span class="header-section-number">2</span> Ejemplo sencillo: Gatos y horas de sueño</h1>
<p><img src="https://images.unsplash.com/photo-1541781774459-bb2af2f05b55?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&amp;ixlib=rb-1.2.1&amp;auto=format&amp;fit=crop&amp;w=1042&amp;q=80" /> Recordado el ejemplo de nuestro tema anterior, estos datos ficticios corresponden a valores diarios del nivel de irritabilidad de una persona, las horas de sueño de esa persona y las horas de sueño de su gato (el ejemplo original de <a href="https://learningstatisticswithr-bookdown.netlify.app/regression.html#multipleregression">Navarro</a> usa las horas de sueño de un bebe, pero los gatos son mas lindos).</p>
<p>En nuestro tema anterior, construimos un modelo lineal simple para predecir los niveles de irritabildiad en función de las horas de sueño, y otro modelo diferente para predecir la irritabilidad en función de las horas de sueño del gato.</p>
<p>Sin embargo, ambas variables van a afectar la irritabilidad por lo que tiene sentido generar un modelo de regresión múltiple que considere ambas variable.</p>
<pre class="r"><code>#abrimos los datos
library(tidyverse)


gatos_url &lt;- &quot;https://raw.githubusercontent.com/trippv/Miguel_Tripp/master/gatos_sueno.csv&quot;

gatos &lt;- read_csv(gatos_url)</code></pre>
<p>En R, para generar un modelo multiple simplemente agregamos mas predictores en nuestra formula de la siguiente manera</p>
<pre class="r"><code>Regresion.multi &lt;- lm(irritabilidad ~ horas_sueno + sueno_gatos, data = gatos)</code></pre>
<p>Al igual que en el tema anterior, podemos visualizar los coeficientes estimados por R</p>
<pre class="r"><code>print(Regresion.multi)</code></pre>
<pre><code>
Call:
lm(formula = irritabilidad ~ horas_sueno + sueno_gatos, data = gatos)

Coefficients:
(Intercept)  horas_sueno  sueno_gatos  
  125.96557     -8.95025      0.01052  </code></pre>
<p>De esta manera, nuestra formula de regresión quedaria de la siguiente forma:</p>
<p><span class="math inline">\(Y=\beta_0 + \beta_1 horas~de~sueno + \beta_2 sueno~ gato + \epsilon\)</span></p>
<p>De acuerdo a los resultados, el coeficiente asociado a las horas de sueño <code>horas_sueno</code>es bastante alto, lo que indica que <strong>cada hora de sueño que pierda que hace mas irritable</strong> mientras que el coeficiente del gatos <code>sueno_gato</code> es muy pequeño, sugiriendo que no tiene mucho efecto en mi irritabilidad.</p>
<p>Utilizando <code>summary()</code> podemos acceder a las pruebas de significancia del modelo.</p>
<pre class="r"><code>summary(Regresion.multi)</code></pre>
<pre><code>
Call:
lm(formula = irritabilidad ~ horas_sueno + sueno_gatos, data = gatos)

Residuals:
     Min       1Q   Median       3Q      Max 
-11.0345  -2.2198  -0.4016   2.6775  11.7496 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) 125.96557    3.04095  41.423   &lt;2e-16 ***
horas_sueno  -8.95025    0.55346 -16.172   &lt;2e-16 ***
sueno_gatos   0.01052    0.27106   0.039    0.969    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 4.354 on 97 degrees of freedom
Multiple R-squared:  0.8161,    Adjusted R-squared:  0.8123 
F-statistic: 215.2 on 2 and 97 DF,  p-value: &lt; 2.2e-16</code></pre>
<p>En la tabla, cada fila corresponde a cada uno de los coeficientes estimados y se realiza una prueba T de Student para evlauar si son diferentes de 0.</p>
<p>Finalmente, los resultados nos muestra que el modelo se desempeña mehor que lo esperado solamente por efecto del azar (<span class="math inline">\(F(2,97) = 215.2, P &lt; 0.001\)</span>) y el coeficiente de determinación <span class="math inline">\(R^2 = 0.812\)</span> nos indica que el modelo de regresión nos explica el 81.2% de la variabilidad.</p>
<p>Sin embargo, algo interesante es que el coeficiente <code>sueno_gatos</code> no es significativo, lo que nos sugiere que posiblemente este modelo no sea el mas eficiente para predecir nuestra irritabildiad y que podriamos eliminar este predictor.</p>
</div>
<div id="ejemplo-complicado-predecir-el-peso-de-un-pez" class="section level1" number="3">
<h1 number="3"><span class="header-section-number">3</span> Ejemplo complicado: Predecir el peso de un pez</h1>
<p><img src="https://upload.wikimedia.org/wikipedia/commons/thumb/b/ba/0418Blackchin_tilapia_Gloria_fish_textures_03.jpg/640px-0418Blackchin_tilapia_Gloria_fish_textures_03.jpg" /></p>
<p>Estos datos estan tomados y modificados del set de datos <code>Fish Market</code> disponible en <a href="https://www.kaggle.com/aungpyaeap/fish-market">kaggle</a>. Para este ejercicio, solo utilizaremos los datos de <strong>tilapia</strong></p>
<pre class="r"><code>peces_url &lt;- &quot;https://raw.githubusercontent.com/trippv/Miguel_Tripp/master/peces_regresion.csv&quot;

peces &lt;- read_csv(peces_url)</code></pre>
<pre><code>
-- Column specification --------------------------------------------------------
cols(
  Especie = col_character(),
  Peso = col_double(),
  Longitud1 = col_double(),
  Longitud2 = col_double(),
  Longitud3 = col_double(),
  Longitud4 = col_double(),
  Alto = col_double(),
  Ancho = col_double()
)</code></pre>
<div id="htmlwidget-68fb61e564b238be9b65" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-68fb61e564b238be9b65">{"x":{"filter":"none","extensions":["Buttons"],"data":[["Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Tilapia","Rutilo","Rutilo","Rutilo","Rutilo","Rutilo","Rutilo","Rutilo","Rutilo","Rutilo","Rutilo","Rutilo","Rutilo","Rutilo","Rutilo","Rutilo","Rutilo","Rutilo","Rutilo","Rutilo","Rutilo","salmon","salmon","salmon","salmon","salmon","salmon","Parkki","Parkki","Parkki","Parkki","Parkki","Parkki","Parkki","Parkki","Parkki","Parkki","Parkki","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca","Perca"],[242,290,340,363,430,450,500,390,450,500,475,500,500,340,600,600,700,700,610,650,575,685,620,680,700,725,720,714,850,1000,920,955,925,975,950,40,69,78,87,120,0,110,120,150,145,160,140,160,169,161,200,180,290,272,390,270,270,306,540,800,1000,55,60,90,120,150,140,170,145,200,273,300,5.9,32,40,51.5,70,100,78,80,85,85,110,115,125,130,120,120,130,135,110,130,150,145,150,170,225,145,188,180,197,218,300,260,265,250,250,300,320,514,556,840,685,700,700,690,900,650,820,850,900,1015,820,1100,1000,1100,1000,1000],[23.2,24,23.9,26.3,26.5,26.8,26.8,27.6,27.6,28.5,28.4,28.7,29.1,29.5,29.4,29.4,30.4,30.4,30.9,31,31.3,31.4,31.5,31.8,31.9,31.8,32,32.7,32.8,33.5,35,35,36.2,37.4,38,12.9,16.5,17.5,18.2,18.6,19,19.1,19.4,20.4,20.5,20.5,21,21.1,22,22,22.1,23.6,24,25,29.5,23.6,24.1,25.6,28.5,33.7,37.3,13.5,14.3,16.3,17.5,18.4,19,19,19.8,21.2,23,24,7.5,12.5,13.8,15,15.7,16.2,16.8,17.2,17.8,18.2,19,19,19,19.3,20,20,20,20,20,20.5,20.5,20.7,21,21.5,22,22,22.6,23,23.5,25,25.2,25.4,25.4,25.4,25.9,26.9,27.8,30.5,32,32.5,34,34,34.5,34.6,36.5,36.5,36.6,36.9,37,37,37.1,39,39.8,40.1,40.2,41.1],[25.4,26.3,26.5,29,29,29.7,29.7,30,30,30.7,31,31,31.5,32,32,32,33,33,33.5,33.5,34,34,34.5,35,35,35,35,36,36,37,38.5,38.5,39.5,41,41,14.1,18.2,18.8,19.8,20,20.5,20.8,21,22,22,22.5,22.5,22.5,24,23.4,23.5,25.2,26,27,31.7,26,26.5,28,31,36.4,40,14.7,15.5,17.7,19,20,20.7,20.7,21.5,23,25,26,8.4,13.7,15,16.2,17.4,18,18.7,19,19.6,20,21,21,21,21.3,22,22,22,22,22,22.5,22.5,22.7,23,23.5,24,24,24.6,25,25.6,26.5,27.3,27.5,27.5,27.5,28,28.7,30,32.8,34.5,35,36.5,36,37,37,39,39,39,40,40,40,40,42,43,43,43.5,44],[30,31.2,31.1,33.5,34,34.7,34.5,35,35.1,36.2,36.2,36.2,36.4,37.3,37.2,37.2,38.3,38.5,38.6,38.7,39.5,39.2,39.7,40.6,40.5,40.9,40.6,41.5,41.6,42.6,44.1,44,45.3,45.9,46.5,16.2,20.3,21.2,22.2,22.2,22.8,23.1,23.7,24.7,24.3,25.3,25,25,27.2,26.7,26.8,27.9,29.2,30.6,35,28.7,29.3,30.8,34,39.6,43.5,16.5,17.4,19.8,21.3,22.4,23.2,23.2,24.1,25.8,28,29,8.8,14.7,16,17.2,18.5,19.2,19.4,20.2,20.8,21,22.5,22.5,22.5,22.8,23.5,23.5,23.5,23.5,23.5,24,24,24.2,24.5,25,25.5,25.5,26.2,26.5,27,28,28.7,28.9,28.9,28.9,29.4,30.1,31.6,34,36.5,37.3,39,38.3,39.4,39.3,41.4,41.4,41.3,42.3,42.5,42.4,42.5,44.6,45.2,45.5,46,46.6],[12.52,16.48,16.3778,13.73,14.444,19.6024,21.1795,14.67,18.0049,21.2266,19.2628,17.3714,14.7592,15.9129,18.9544,20.438,16.8604,16.938,17.633,19.4738,20.1285,17.9936,17.5227,21.4686,22.2405,17.36,18.3618,19.517,21.8896,23.957,24.0369,24.084,21.7542,20.6354,21.6235,9.1472,11.2983,12.5756,11.6166,11.216,12.4752,13.1677,8.1146,6.8045,7.6339,8.0334,12.55,11.4,12.5344,8.9153,13.3968,14.0866,12.8768,14.568,15.485,15.3804,10.1454,9.778,17.744,13.7612,17.354,7.8475,10.5772,11.4052,12.3922,9.8928,12.5376,10.396,13.7364,13.3458,17.088,18.368,5.112,9.528,4.824,6.5924,5.588,10.2224,6.1992,11.6358,8.1376,8.082,8.6925,11.9175,8.6925,13.384,10.11,11.64,8.11,12.875,7.5225,12.856,12.792,12.9532,9.2185,8.275,10.293,12.375,9.7334,8.4395,11.561,8.168,14.323,12.1672,13.0516,14.2828,8.8204,14.5852,9.6156,11.03,11.2565,12.4884,11.881,14.6091,14.835,11.5717,18.1366,15.1366,18.4313,13.9286,16.73,18.3808,14.135,15.8002,18.9328,13.5125,18.604,19.4888],[11.52,12.48,12.3778,12.73,12.444,13.6024,14.1795,12.67,14.0049,14.2266,14.2628,14.3714,13.7592,13.9129,14.9544,15.438,14.8604,14.938,15.633,14.4738,15.1285,15.9936,15.5227,15.4686,16.2405,16.36,16.3618,16.517,16.8896,18.957,18.0369,18.084,18.7542,18.6354,17.6235,4.1472,5.2983,5.5756,5.6166,6.216,6.4752,6.1677,6.1146,5.8045,6.6339,7.0334,6.55,6.4,7.5344,6.9153,7.3968,7.0866,8.8768,8.568,9.485,8.3804,8.1454,8.778,10.744,11.7612,12.354,6.8475,6.5772,7.4052,8.3922,8.8928,8.5376,9.396,9.7364,10.3458,11.088,11.368,2.112,3.528,3.824,4.5924,4.588,5.2224,5.1992,5.6358,5.1376,5.082,5.6925,5.9175,5.6925,6.384,6.11,5.64,6.11,5.875,5.5225,5.856,6.792,5.9532,5.2185,6.275,7.293,6.375,6.7334,6.4395,6.561,7.168,8.323,7.1672,7.0516,7.2828,7.8204,7.5852,7.6156,10.03,10.2565,11.4884,10.881,10.6091,10.835,10.5717,11.1366,11.1366,12.4313,11.9286,11.73,12.3808,11.135,12.8002,11.9328,12.5125,12.604,12.4888],[4.02,4.3056,4.6961,4.4555,5.134,4.9274,5.2785,4.69,4.8438,4.9594,5.1042,4.8146,4.368,5.0728,5.1708,5.58,5.2854,5.1975,5.1338,5.7276,5.5695,5.3704,5.2801,6.1306,5.589,6.0532,6.09,5.8515,6.1984,6.603,6.3063,6.292,6.7497,6.7473,6.3705,2.268,2.8217,2.9044,3.1746,3.5742,3.3516,3.3957,3.2943,3.7544,3.5478,3.8203,3.325,3.8,3.8352,3.6312,4.1272,3.906,4.4968,4.7736,5.355,4.2476,4.2485,4.6816,6.562,6.5736,6.525,2.3265,2.3142,2.673,2.9181,3.2928,3.2944,3.4104,3.1571,3.6636,4.144,4.234,1.408,1.9992,2.432,2.6316,2.9415,3.3216,3.1234,3.0502,3.0368,2.772,3.555,3.3075,3.6675,3.534,3.4075,3.525,3.525,3.525,3.995,3.624,3.624,3.63,3.626,3.725,3.723,3.825,4.1658,3.6835,4.239,4.144,5.1373,4.335,4.335,4.5662,4.2042,4.6354,4.7716,6.018,6.3875,7.7957,6.864,6.7408,6.2646,6.3666,7.4934,6.003,7.3514,7.1064,7.225,7.4624,6.63,6.8684,7.2772,7.4165,8.142,7.5958]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Especie<\/th>\n      <th>Peso<\/th>\n      <th>Longitud1<\/th>\n      <th>Longitud2<\/th>\n      <th>Longitud3<\/th>\n      <th>Longitud4<\/th>\n      <th>Alto<\/th>\n      <th>Ancho<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","buttons":[{"extend":"collection","buttons":"copy","text":"Copiar"},{"extend":"collection","buttons":"csv","text":"Descargar"}],"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>tilapia &lt;- peces[peces$Especie == &quot;Tilapia&quot;, - 1]</code></pre>
<p>El primer paso para establecer un modelo lineal múltiple es estudiar la relación que existe entre variables. Esta relación en critica a la hora de identificar los mejores predictores para el modelo (p. ej, que variables presentan relación de tipo no lineal) y para identificar colinialidad entre predictores.</p>
<p>Podemos usar la funcion <code>pairs.panel()</code> del paquete <code>psych</code> para visualziar los diagramas de disperción entre todas las variables asi como su valor de correlación.</p>
<pre class="r"><code>psych::pairs.panels(tilapia, stars = TRUE)</code></pre>
<p><img src="figure/Clase12_regresionMulti.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>El paquete Psych tambien contien la función <code>multi.hist()</code>la cual nos ayuda a evaluar la distribución de los datos:</p>
<pre class="r"><code>psych::multi.hist(x = tilapia[,-1], dcol = c(&quot;blue&quot;, &quot;red&quot;))</code></pre>
<p><img src="figure/Clase12_regresionMulti.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Donde la linea roja corresponde a la distribución normal ajustada y la linea azul a la densidad observada.</p>
<p>Del análisis preliminar se pueden extraer las siguientes conclusiones:</p>
<ul>
<li>Las variables que tienen mayor relación lineal con el peso son: ancho (r = 0.96), longitud3 y longitud2 (r = 0.95).</li>
<li>Las longitudes 1,2 y 3 estan altamente correlacionadas entre ellas, por lo que posiblemente no sea útil introducir ambos predictores en el modelo.</li>
</ul>
<div id="generar-el-modelo-multiple" class="section level2" number="3.1">
<h2 number="3.1"><span class="header-section-number">3.1</span> Generar el modelo multiple</h2>
<p>Al igual que en nuestro ejemplo mas simple, el modelo múltiple se genera añadiendo cada uno de los predictores a la ecuación de regresión utilizando el simbolo <code>+</code>.</p>
<p>De manera que podemos pasar de un modelo simple usando un solo predictor, por ejemplo <code>Longitud1</code></p>
<pre class="r"><code>tilapia_lm1 &lt;- lm(Peso ~ Longitud1, data = tilapia  )</code></pre>
<p>A un modelo mas complejo utilizando todas las variables:</p>
<pre class="r"><code>tilapia_lm2 &lt;- lm(Peso ~ Longitud1 + Longitud2 + Longitud3 + Longitud4 + Ancho + Alto, data = tilapia)</code></pre>
<style>
div.blue { background-color:#e6f0ff; border-radius: 5px; padding: 20px;}
</style>
<div class="blue">
<p>Una forma fácil de escribir una ecuación multiple cuando se van a utilizar <em>todas</em> las variables de una tablas es utilizando el <code>.</code>de la siguiente forma</p>
<pre class="r"><code>tilapia_lm2 &lt;- lm(Peso ~ ., data = tilapia)</code></pre>
</div>
<p>Utilizando la función <code>summary()</code> podemos comparar ambos modelos</p>
<ol style="list-style-type: decimal">
<li>Modelo con un predictor:</li>
</ol>
<pre class="r"><code>summary(tilapia_lm1)</code></pre>
<pre><code>
Call:
lm(formula = Peso ~ Longitud1, data = tilapia)

Residuals:
     Min       1Q   Median       3Q      Max 
-233.877  -35.377   -4.797   31.578  207.923 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -1035.348    107.973  -9.589 4.58e-11 ***
Longitud1      54.550      3.539  15.415  &lt; 2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 74.15 on 33 degrees of freedom
Multiple R-squared:  0.8781,    Adjusted R-squared:  0.8744 
F-statistic: 237.6 on 1 and 33 DF,  p-value: &lt; 2.2e-16</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Modelo con todos los predictores:</li>
</ol>
<pre class="r"><code>summary(tilapia_lm2)</code></pre>
<pre><code>
Call:
lm(formula = Peso ~ Longitud1 + Longitud2 + Longitud3 + Longitud4 + 
    Ancho + Alto, data = tilapia)

Residuals:
     Min       1Q   Median       3Q      Max 
-164.544  -27.502   -4.932   23.936  115.699 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -939.0144   165.4496  -5.676  4.4e-06 ***
Longitud1     16.4030    50.2746   0.326  0.74665    
Longitud2     18.9733    55.0268   0.345  0.73282    
Longitud3    -21.9432    43.2725  -0.507  0.61606    
Longitud4      0.8675     5.6808   0.153  0.87972    
Ancho         55.4062    44.2011   1.254  0.22039    
Alto          62.9731    19.8432   3.174  0.00364 ** 
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 54.89 on 28 degrees of freedom
Multiple R-squared:  0.9433,    Adjusted R-squared:  0.9312 
F-statistic: 77.64 on 6 and 28 DF,  p-value: 3.817e-16</code></pre>
<p>Una de las primeras cosas que observamos es la diferencia entre los valores de <span class="math inline">\(R^2\)</span> el cual es una medida de la variabilidad en el resultado es explicada por los predictores. Cuando solo utilizamos el valor de la <code>Longitud1</code>, el valor de <span class="math inline">\(R^2\)</span> es de 0.878 lo que indica que la Longitud 1 representa el 87.8% de la variación en el peso de las tilapias. Por otro lado, cuando el resto de los predictores son utilizados, el valor de <span class="math inline">\(R^2\)</span> se incrementa a 0.943, por lo que la inclusión de estos predictores representa un incremento adicional de la variación explicada.</p>
<p>Sin embargo, es importante notar que <strong>todas las variables menos <code>Alto</code> no son significativas</strong>, lo que es un indicativo de que estas variables podrian no estar contribuyendo al modelo.</p>
<pre class="r"><code>tilapia_lm3 &lt;- lm(Peso ~ Alto, data = tilapia)

summary(tilapia_lm3)</code></pre>
<pre><code>
Call:
lm(formula = Peso ~ Alto, data = tilapia)

Residuals:
     Min       1Q   Median       3Q      Max 
-147.362  -35.831   -5.472   29.357  115.326 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -941.559     74.896  -12.57 3.93e-14 ***
Alto         102.705      4.893   20.99  &lt; 2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 56.06 on 33 degrees of freedom
Multiple R-squared:  0.9303,    Adjusted R-squared:  0.9282 
F-statistic: 440.5 on 1 and 33 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<div id="selección-de-los-mejores-predictores" class="section level2" number="3.2">
<h2 number="3.2"><span class="header-section-number">3.2</span> Selección de los mejores predictores</h2>
<p>Cuando construimos un modelo de regresión múltiple, una de las preguntas mas obvias es cuales predictores utilizar. En el ejemplo anterior vimos que aparenmente no todos los predictores son en realidad informativos en el modelo. En este ejemplo en partiucular esto puede estar relacionado con la colinealidad de los datos.</p>
<p>En un mundo ideal, los predictores deben seleccionarse a partir de información prexistente. Si un nuevo modelo se va a añadir a un modelo existente, entonces este predictir debe tener sustento teórico. Algo que definitivamente se debe de evitar es añadir cientos de predictores al azar a un modelo y esperar que ocurra lo mejor.</p>
<p>Un método (de tantos) para seleccionar nuestro predictores es utlizando un método <em>stepwise</em> el cual basa la elección de los predictores basados en criterios matemáticos.</p>
<p>La función <code>step()</code> permite evaluar el poder predictivo de un modelo al añadir (<em>forward</em>) o quitar (<em>backward</em>) predictores a un modelo inicial hasta encontrar los mejores predictores basandose en la estimación del <strong>Criterio de Información de Akaike (AIC)</strong>.</p>
<style>
div.blue { background-color:#e6f0ff; border-radius: 5px; padding: 20px;}
</style>
<div class="blue">
<p>Puedes leer mas sobre el AIC <a href="https://es.wikipedia.org/wiki/Criterio_de_informaci%C3%B3n_de_Akaike">aquí</a>, pero de manera general, a mayor AIC el modelo es peor y a menor AIC el modelo el mejor.</p>
</div>
<pre class="r"><code>step(tilapia_lm2, direction = &quot;backward&quot;, trace = 1)</code></pre>
<pre><code>Start:  AIC=286.57
Peso ~ Longitud1 + Longitud2 + Longitud3 + Longitud4 + Ancho + 
    Alto

            Df Sum of Sq    RSS    AIC
- Longitud4  1      70.3  84438 284.60
- Longitud1  1     320.8  84688 284.70
- Longitud2  1     358.2  84726 284.71
- Longitud3  1     774.8  85143 284.89
- Ancho      1    4734.4  89102 286.48
&lt;none&gt;                    84368 286.57
- Alto       1   30346.3 114714 295.32

Step:  AIC=284.59
Peso ~ Longitud1 + Longitud2 + Longitud3 + Ancho + Alto

            Df Sum of Sq    RSS    AIC
- Longitud1  1       313  84751 282.72
- Longitud2  1       326  84764 282.73
- Longitud3  1       732  85170 282.90
&lt;none&gt;                    84438 284.60
- Ancho      1      5337  89775 284.74
- Alto       1     37700 122138 295.51

Step:  AIC=282.72
Peso ~ Longitud2 + Longitud3 + Ancho + Alto

            Df Sum of Sq    RSS    AIC
- Longitud3  1       494  85245 280.93
- Longitud2  1      1555  86306 281.36
&lt;none&gt;                    84751 282.72
- Ancho      1      5461  90211 282.91
- Alto       1     39035 123786 293.98

Step:  AIC=280.93
Peso ~ Longitud2 + Ancho + Alto

            Df Sum of Sq    RSS    AIC
- Ancho      1      5000  90244 280.92
&lt;none&gt;                    85245 280.93
- Longitud2  1      8012  93256 282.07
- Alto       1     38542 123787 291.98

Step:  AIC=280.92
Peso ~ Longitud2 + Alto

            Df Sum of Sq    RSS    AIC
&lt;none&gt;                    90244 280.92
- Longitud2  1     13455 103699 283.79
- Alto       1     65408 155652 298.00</code></pre>
<pre><code>
Call:
lm(formula = Peso ~ Longitud2 + Alto, data = tilapia)

Coefficients:
(Intercept)    Longitud2         Alto  
   -1013.12        16.35        71.77  </code></pre>
<p>Con esta información, podemos generar un nuevo modelo con los predictores indicados con la función step</p>
<pre class="r"><code>tilapia_lm4 &lt;- lm(Peso ~ Alto + Longitud2, data = tilapia)

summary(tilapia_lm4)</code></pre>
<pre><code>
Call:
lm(formula = Peso ~ Alto + Longitud2, data = tilapia)

Residuals:
     Min       1Q   Median       3Q      Max 
-168.537  -24.862   -5.562   22.370  107.114 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -1013.115     78.150 -12.964 2.79e-14 ***
Alto           71.769     14.902   4.816 3.39e-05 ***
Longitud2      16.348      7.485   2.184   0.0364 *  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 53.1 on 32 degrees of freedom
Multiple R-squared:  0.9394,    Adjusted R-squared:  0.9356 
F-statistic: 247.8 on 2 and 32 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<div id="comparar-dos-modelos" class="section level2" number="3.3">
<h2 number="3.3"><span class="header-section-number">3.3</span> Comparar dos modelos</h2>
<p>Para comparar el ajuste de dos modelos, es posible utilizar la función <code>anova()</code> con los modelos de regresión generados en forma de dos o mas argumentos. Esta función tomará los modelos como dos dos argumentos y probará si un modelo mas complejo es significativamente mejor capturando la variabilidad de los datos que un modelo mas sencillo. Si el valor P resultante es significativo, entonces concluimos que el modelo mas complejo es significativamente mejor qu el modelo sencillo, y por lo tanto, favorecemos el modelo mas complejo. Si por el contrario el valor P no es significativo, entonces concluimos que el modelo mas complejo no aporta mas información que el modelo sencillo y por lo tanto favorecemos el modelo sencillo.</p>
<p>Entonces, vamos a utilizar la función <code>anova()</code> para comparar el modelo con todos los predictores (<em>tilapia_lm3</em>) y el modelo que solo considera la <strong>altura</strong> y la <strong>longitud2</strong> (<em>tilapia_lm4</em>) y el modelo que solo considera la <strong>altura</strong> (<strong>tilapia_lm2</strong>)</p>
<p>Dentro de la función, se incluyen los modelos del mas sencillo al mas complejo:</p>
<pre class="r"><code>anova(tilapia_lm3, tilapia_lm4, tilapia_lm2)</code></pre>
<pre><code>Analysis of Variance Table

Model 1: Peso ~ Alto
Model 2: Peso ~ Alto + Longitud2
Model 3: Peso ~ Longitud1 + Longitud2 + Longitud3 + Longitud4 + Ancho + 
    Alto
  Res.Df    RSS Df Sum of Sq      F  Pr(&gt;F)  
1     33 103699                              
2     32  90244  1   13454.8 4.4654 0.04363 *
3     28  84368  4    5876.7 0.4876 0.74473  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>De los resultados de esta prueba podemos concluir que:</p>
<ol style="list-style-type: decimal">
<li>La comparación del modelo mas sencillo con un solo predictor con el modelo con dos predictores es significativa, lo que indica que favorecemos al modelo 2</li>
<li>La comparación del model con dos predictores con el modelo con todos los predictores no es significativa, por lo que favorecemos el modelo con dos predictores.</li>
</ol>
<p>Cada una de las pendientes de un modelo de regresión lineal múltiple (coeficientes parciales de regresión de los predictores) se define del siguiente modo: Si el resto de variables se mantienen constantes, por cada unidad que aumenta el predictor en cuestión, la variable (<span class="math inline">\(Y\)</span>) varía en promedio tantas unidades como indica la pendiente. Para este ejemplo, <strong>por cada unidad que aumenta el predictor Alto, el peso aumenta en promedio 71.76 unidades, manteniéndose constantes el resto de predictores.</strong></p>
</div>
<div id="validación-de-las-condiciones-para-la-regresion-lineal-multiple" class="section level2" number="3.4">
<h2 number="3.4"><span class="header-section-number">3.4</span> Validación de las condiciones para la regresion lineal multiple</h2>
<p>La relación lieal entre los predictores numéricos y la variable de respuesta se puede evaluar con diagramas de dispersión entre la variable dependiente y cada uno de los predictores o con diagramas de dispersión entre cada uno de los predictores y los residuos del modelo. Si la relación es lineal, los residuos deben de distribuirse aleatoriamente en torno a 0 con una variabilidad constante a lo largo del eje X.</p>
<pre class="r"><code>library(patchwork)
Alto_res &lt;- ggplot(tilapia, aes(x = Alto, y = tilapia_lm4$residuals))+
  geom_point()+
  geom_smooth(color = &quot;firebrick&quot;)+
  geom_hline(yintercept = 0)+
  theme_bw()

Longitud2_res &lt;- ggplot(tilapia, aes(x = Longitud2, y = tilapia_lm4$residuals))+
  geom_point()+
  geom_smooth(color = &quot;firebrick&quot;)+
  geom_hline(yintercept = 0)+
  theme_bw()

Alto_res + Longitud2_res</code></pre>
<pre><code>`geom_smooth()` using method = &#39;loess&#39; and formula &#39;y ~ x&#39;
`geom_smooth()` using method = &#39;loess&#39; and formula &#39;y ~ x&#39;</code></pre>
<p><img src="figure/Clase12_regresionMulti.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow = c(2,2))
plot(tilapia_lm4)</code></pre>
<p><img src="figure/Clase12_regresionMulti.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
</div>
<div id="conclusión" class="section level2" number="3.5">
<h2 number="3.5"><span class="header-section-number">3.5</span> Conclusión</h2>
<p>El modelo:</p>
<p><span class="math inline">\(Peso = −1013.115+ 71.76~Alto+16.34~Longitud2\)</span></p>
<p>es capaz de explicar el 94% de la variabilidad observada en el peso. El test <span class="math inline">\(F\)</span> muestra que es significativo (<span class="math inline">\(P&lt;2.22^{−16}\)</span>).</p>
<p>Tambien es posible utilizar la función <code>report()</code> para generar un reporte fácil de interpretar</p>
<pre class="r"><code>report::report(tilapia_lm4)</code></pre>
<pre><code>We fitted a linear model (estimated using OLS) to predict Peso with Alto and Longitud2 (formula: Peso ~ Alto + Longitud2). The model explains a statistically significant and substantial proportion of variance (R2 = 0.94, F(2, 32) = 247.83, p &lt; .001, adj. R2 = 0.94). The model&#39;s intercept, corresponding to Alto = 0 and Longitud2 = 0, is at -1013.12 (95% CI [-1172.30, -853.93], t(32) = -12.96, p &lt; .001). Within this model:

  - The effect of Alto is statistically significant and positive (beta = 71.77, 95% CI [41.41, 102.12], t(32) = 4.82, p &lt; .001; Std. beta = 0.67, 95% CI [0.39, 0.96])
  - The effect of Longitud2 is statistically significant and positive (beta = 16.35, 95% CI [1.10, 31.59], t(32) = 2.18, p = 0.036; Std. beta = 0.31, 95% CI [0.02, 0.59])

Standardized parameters were obtained by fitting the model on a standardized version of the dataset.</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19043)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] patchwork_1.1.1 forcats_0.5.1   stringr_1.4.0   dplyr_1.0.5    
 [5] purrr_0.3.4     readr_1.4.0     tidyr_1.1.3     tibble_3.0.4   
 [9] ggplot2_3.3.5   tidyverse_1.3.1 workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] nlme_3.1-152      fs_1.5.0          lubridate_1.7.10  insight_0.14.2   
 [5] httr_1.4.2        rprojroot_2.0.2   tools_4.0.5       backports_1.2.1  
 [9] bslib_0.2.5.1     utf8_1.2.1        R6_2.5.0          DT_0.17          
[13] DBI_1.1.0         mgcv_1.8-33       colorspace_2.0-0  withr_2.4.2      
[17] tidyselect_1.1.1  mnormt_2.0.2      emmeans_1.5.3     curl_4.3         
[21] compiler_4.0.5    git2r_0.27.1      performance_0.7.2 cli_2.5.0        
[25] rvest_1.0.0       xml2_1.3.2        sandwich_3.0-0    labeling_0.4.2   
[29] bayestestR_0.10.0 sass_0.4.0        scales_1.1.1      mvtnorm_1.1-1    
[33] psych_2.0.12      digest_0.6.27     rmarkdown_2.9     pkgconfig_2.0.3  
[37] htmltools_0.5.1.1 dbplyr_2.1.1      htmlwidgets_1.5.3 rlang_0.4.11     
[41] readxl_1.3.1      rstudioapi_0.13   jquerylib_0.1.4   generics_0.1.0   
[45] farver_2.0.3      zoo_1.8-8         jsonlite_1.7.2    crosstalk_1.1.0.1
[49] magrittr_2.0.1    parameters_0.14.0 Matrix_1.3-2      Rcpp_1.0.5       
[53] munsell_0.5.0     fansi_0.4.2       lifecycle_1.0.0   multcomp_1.4-17  
[57] stringi_1.5.3     whisker_0.4       yaml_2.2.1        MASS_7.3-53      
[61] grid_4.0.5        parallel_4.0.5    promises_1.1.1    crayon_1.4.1     
[65] lattice_0.20-41   haven_2.3.1       splines_4.0.5     hms_1.0.0        
[69] tmvnsim_1.0-2     knitr_1.30        ps_1.5.0          pillar_1.6.0     
[73] estimability_1.3  effectsize_0.4.5  codetools_0.2-18  reprex_2.0.0     
[77] glue_1.4.2        evaluate_0.14     modelr_0.1.8      vctrs_0.3.8      
[81] httpuv_1.5.4      cellranger_1.1.0  gtable_0.3.0      assertthat_0.2.1 
[85] xfun_0.23         xtable_1.8-4      broom_0.7.6       coda_0.19-4      
[89] later_1.1.0.1     survival_3.2-10   report_0.3.5      TH.data_1.0-10   
[93] ellipsis_0.3.1   </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = false;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
